src/apps/FlappyBird.jsx

```diff
@@ -0,0 +1,362 @@
+    const sWid = ctx.measureText(hScore).width;
+    const sY = 80;
+
+    ctx.fillStyle = "white";
+    ctx.fillText(score, 20, floor + sY);
+    ctx.fillText(hScore, wid - sWid - 20, floor + sY);
+  };
+
+  const adjust = () => {
+    if (trans % pillSpace === 0) {
+      let h;
+      pillars.push({
+        x: -trans + wid,
+        y: 0,
+        w: pillWid,
+        h: (h = Math.random() * (floor - 300) + 100),
+      });
+
+      pillars.push({
+        x: -trans + wid,
+        y: h + pillGap,
+        w: pillWid,
+        h: floor - h - pillGap,
+      });
+    }
+
+    pillars = pillars.filter((pill) => -trans < pill.x + pill.w);
+
+    player.v += gravity;
+    if (player.v > termVel) {
+      player.v = termVel;
+    }
+    player.y += player.v;
+
+    if (player.y < player.r) {
+      player.y = player.r;
+      player.v = 0;
+    }
+
+    for (const i in pillars) {
+      let pillIndex = i;
+      const pill = pillars[pillIndex];
+      if (
+        pill.x + trans < player.x + player.r &&
+        pill.x + pill.w + trans > player.x - player.r
+      ) {
+        if (
+          player.y - player.r > pill.y &&
+          player.y - player.r < pill.y + pill.h
+        ) {
+          colider = true;
+          running = false;
+          render();
+          break;
+        }
+        if (
+          player.y + player.r < pill.y + pill.h &&
+          player.y + player.r > pill.y
+        ) {
+          colider = true;
+          running = false;
+          render();
+          break;
+        }
+        if (!pill.passed && pillIndex % 2 == 1) {
+          score += 1;
+          pill.passed = true;
+        }
+      }
+    }
+
+    if (player.y + player.r - player.v > floor) {
+      player.y = floor - player.r;
+      running = false;
+      colider = true;
+      render();
+    }
+  };
+
+  document.onmousedown = () => {
+    if (running) {
+      player.v = thrust;
+    } else if (!colider) {
+      running = true;
+    } else {
+      if (score > high) {
+        localStorage.setItem("high", score);
+      }
+      init();
+    }
+  };
+  init();
+  function loop() {
+    if (running) {
+      adjust();
+    }
+    if (!colider) {
+      render();
+    }
+    requestAnimationFrame(loop);
+  }
+};
+
+function Renderer() {
+  useEffect(() => {
+    gameLogic();
+  }, []);
+
+  return (
+    <div>
+      <NyanComponentRender />
+    </div>
+  );
+}
+
+const shouldRunMyApp = async ({ query }) => {
+  const triggerQueries = ["i am bored", "flappy bird", "nyan cat flappy bird"];
+
+  for (const triggerQuery of triggerQueries) {
+    if (query.toLowerCase() === triggerQuery) {
+      return { query };
+    }
+  }
+
+  return;
+};
+
+const MyApp = {
+  name: "FlappyBird",
+  description: "A flappy bird game for when you are bored!",
+  // queryToData takes in the query and returns data which
+  // the Component displays on the website.
+  // If queryToData returns no data, we do not display the app.
+  queryToData: shouldRunMyApp,
```

Comment:
We made some utility functions to save time writing functions regarding query matching in a app. You can use the function called `matchTriggerQueries` directly, like it has been used here - https://github.com/felvin-search/instant-apps/blob/master/src/apps/WordCounter.tsx#L34-L40